core-bundle/src/Resources/contao/languages/en/tl_page.xlf

```diff
@@ -30,7 +30,7 @@
         <source>Language</source>
       </trans-unit>
       <trans-unit id="tl_page.language.1">
-        <source>Please enter the page language according to the ISO-639-1 standard (e.g. "en" for English or "en-US" for American English).</source>
+        <source>Please enter the page language according to the ISO-639 standard (e.g. "en" for English or "en_US" for American English).</source>
```

Comment:
How do we handle this case? The format is different depending on whether legacy routing is enabled or not. Also, a locale can be much more specific (e.g. `zh_Hant_TW`).

Reply:
> The format is different depending on whether legacy routing is enabled or not. 

In what way?

Reply:
Currently, the format is a language tag (`de-DE`). The new format is an ICU locale (`de_DE`) which can also be extended (`zh_Hant-TW@currency=EUR`). We _could_ always expect/store the new format and convert it automatically, not sure if that would be confusing since we would keep the language tag in URL in legacy mode (new mode has a variable url prefix).

Reply:
IMHO we should continue to use language tags here (so `zh-Hant-TW` instead of `zh_Hant-TW`), because that is what W3 defined for HTML and XML documents: https://www.w3.org/International/articles/language-tags/

Reply:
AFAIR we agreed on using the format to which the `intl` library normalizes to and convert the format if needed for the `lang=""` attribute.

Reply:
I see. That implies that we have to migrate the existing entries and that our users have to learn to use the "old" format again (Contao 4.4 used locales, 4.9 used language tags and 4.12 will be using locales again). Not the best UX …

If we have to convert the format anyway, should we maybe support both locales and language tags?

Reply:
that's already implemented